pkg/handlers/v1/books.go

```diff
@@ -2,56 +2,147 @@ package v1
 
 import (
 	"book_service/pkg/clients"
+	"book_service/pkg/consts"
 	"book_service/pkg/models/common"
 	"book_service/pkg/models/common/req"
+	"book_service/pkg/models/common/res"
 	"book_service/pkg/query"
 	"book_service/pkg/utils"
-	"fmt"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
 	"github.com/gin-gonic/gin"
 	"github.com/google/uuid"
 	"github.com/jinzhu/copier"
-	"github.com/sirupsen/logrus"
-	"net/http"
 )
 
-var booksIndex, _ = utils.GetEnvVar[string]("BOOKS_INDEX", "books")
-
 func GetBookById(c *gin.Context) {
-	payload, _ := c.Get("validated")
-	bookId, _ := payload.(req.GetBook)
-
-	// TODO: it should be in the validation level with extra validators and transformations
-	_, parseErr := uuid.Parse(bookId.ID)
-	if parseErr != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": parseErr})
+	bookReq, err := utils.GetValidatedPayload[req.GetBook](c)
+	if err != nil {
+		log.Errorf("Error getting book by ID: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
 		return
 	}
 
-	esQuery := query.NewQueryBuilder().ID(bookId.ID).Build()
-	fmt.Println(esQuery)
-	hits, err := clients.DoSearch(c, booksIndex, esQuery)
+	esQuery := query.NewQueryBuilder().ID(bookReq.ID).Build()
+	hits, _, err := clients.SearchIndex(c, clients.BooksIndex, esQuery, 1, 0)
```

Comment:
You shouldn't send the client.BooksIndex - this is a const the clients package already have and even more important, this package doesn't need to know about the implementation inside clients.

**Same comment for all the functions in this file**

```suggestion
	hits, _, err := clients.SearchIndex(c, esQuery, 1, 0)
```

Reply:
What will happen if we have 2 indexes?